lacommunaute/forum/models.py

```diff
@@ -86,3 +88,36 @@ def get_stats(self, days_back):
             stats_list[1].reverse()
 
         return stats
+
+    @cached_property
+    def count_engaged_users(self):
+        forums = self.get_family()
+
+        posters = User.objects.filter(posts__topic__forum__in=forums).values_list("email").distinct()
+        likers = User.objects.filter(topic_likes__forum__in=forums).values_list("email").distinct()
+        voters = (
+            User.objects.filter(poll_votes__poll_option__poll__topic__forum__in=forums).values_list("email").distinct()
+        )
+        authenticated_users = posters | likers | voters
+
+        anonymous_posters = (
+            Post.objects.filter(topic__forum__in=forums)
+            .exclude(username__isnull=True)
+            .annotate(email=F("username"))
+            .values_list("email")
```

Comment:
Pas plus direct de faire comme ceci ?
```suggestion
            .values_list("username")
```

Reply:
Les contributions anonymes ne sont pas reliées à un user. Nous enregistrons le `username` et l' `email` dans le `post`. Pour dédoublonner les emails des `user` et des anonymes, j'ai dû faire cette petite coquetterie ^^

Reply:
Donc peut-être rajouter un petit commentaire pour expliquer que c'est voulu ;).